README.md
# Flutter Getting Started Layout

[![Codacy Badge](https://api.codacy.com/project/badge/Grade/25283ed8fd2b4f65b86dd1124632bf2b)](https://app.codacy.com/app/iamriya/GetStarted-FlutterPackage?utm_source=github.com&utm_medium=referral&utm_content=iamriya/GetStarted-FlutterPackage&utm_campaign=Badge_Grade_Dashboard)

A Custom Getting Started Layout with custom Fonts and Colors.

## Preview

![The example app running in Android](https://github.com/MobMaxime/GetStarted-FlutterPackage/blob/master/walkthrough/gif/GettingStarted.gif)

## Installing

1.  Add dependency to `pubspec.yaml`

    *Get the latest version in the 'Installing' tab on [pub.dartlang.org](pub.dartlang.org)*
    
```dart
dependencies:
    get_started: ^0.1.1
```

2.  Import the packages

```dart
import 'package:get_started/classes/walkthrough.dart';
import 'package:get_started/classes/page.dart';
import 'package:get_started/get_started.dart';
import 'package:page_indicator/page_indicator.dart';
```

3.  Adding GetStarted layout

#With required parameters#

```dart
   //In the body of your MaterialApp,
   
    @override
                Widget build(BuildContext context) {
                  return new MaterialApp(
                    debugShowCheckedModeBanner: false,
                    home: Scaffold(
                      //Required Field
                      //body: GetStarted(pageList : pageList),
                      
                      //Required Field with Optional parameters
                      body: GetStarted(pageList : pageList, pageStyle: pageObj, indicatorShape : IndicatorShape.circle(size: 12.0),
                      onFinishPress: _finishPress, onSkipPress: _skipPress),
                    ),
                  );
                }
                
              void _skipPress() {
                    //define your skipButton functionality here.
              }
                
              void _finishPress() {
                    //define your lastNextButton functionality here.
              }
  ```         

#With optional parameters#

```dart
   //All the properties defined for 'page' have some default values. Hence, you may skip some size, etc. if you wish to! But defining a pageObj of page class is a must, so as to pass it as a parameter in GetStarted().
        
          page pageObj = new page(
              bgColor: "#ffffff",
              indicatorColor: "#ffffff",
              indicatorSelectorColor: "#195599",
              nextButtonText: "NEXT",
              nextButtonColor: "#195599",
              nextButtonFont: "Lato",
              nextButtonSize: 20.0,
              skipButtonText: "SKIP",
              skipButtonFont: "Lato",
              skipButtonColor: "#195599",
              skipButtonSize: 20.0,
              lastNextButton:
                  "FINISH", //You may also write "GOT IT" or whatever you wish :)
              lastButtonColor: "#1f8e2f",
              lastButtonFont: "Lato",
              lastButtonSize: 20.0);
              
   //A Page has three sections for Title, Description and an Icon respectively that can be customized for the text, font, color and size.
      
    final List<walkthrough> pageList = [
        walkthrough(
            pageColor: "#48C9B0",
            title: "Welcome!",
            titleFont:
                "Lato", //Don't forget to include yours fonts in pubspec.yaml file.
            titleColor: "#ffffff", //Pass the colours as strings of their hex codes.
            titleSize: 25.0,
            description:
                "This is a Scrollable Text Widget. Hence, don't worry if your description goes long. :) \n\nA Page has three sections for Title, Description and an Icon respectively that can be customized for the text, font, color and size.\n\nAll the properties defined for 'walkthrough' have some default values. Hence, you may skip some size, etc. if you wish to!",
            descriptionFont: "Lato",
            descriptionColor: "#ffffff",
            descriptionSize: 16.0,
            imageIcon: Icon(
              Icons.computer,
              color: Colors.white,
              size: 100.0,
            ))
        ];    
 ```   

## How to use
Check out the **example** app in the [example directory](https://github.com/MobMaxime/GetStarted-FlutterPackage/tree/master/walkthrough/example) or the 'Example' tab on pub.dartlang.org for a more complete example.

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.

## Liked it?
Please star the repo if you liked it or if it helped you somehow! :)


walkthrough/CHANGELOG.md
## \[0.1.1] - June 22, 2019 Release

*   This package contains a Layout screen that can be used as similar to Getting Started Tutorial guide in a Mobile Application.


walkthrough/lib/get_started.dart
library get_started;

import 'package:flutter/material.dart';
import 'package:page_indicator/page_indicator.dart';
import 'classes/walkthrough.dart';
import 'classes/page.dart';

class GetStarted extends StatefulWidget {
  final GestureTapCallback onSkipPress;
  final GestureTapCallback onFinishPress;
  List<Walkthrough> pageList;
  IndicatorShape indicatorShape;
  Page pg = new Page();

  GetStarted({@required List<Walkthrough> pageList, Page pageStyle, IndicatorShape indicatorShape,
      this.onFinishPress, this.onSkipPress}) {
    this.pageList = pageList;
    this.pg = pageStyle == null ? pg : pageStyle;
    this.indicatorShape = indicatorShape == null
        ? IndicatorShape.circle(size: 11.0)
        : indicatorShape;
  }

  @override
  _GetStartedState createState() => _GetStartedState(this.pageList, this.pg,
      this.indicatorShape, this.onFinishPress, this.onSkipPress);
}

class _GetStartedState extends State<GetStarted> {
  PageController pgController = new PageController();
  Page pg;
  GestureTapCallback onSkipPress;
  GestureTapCallback onFinishPress;
  List<Walkthrough> pageList;
  IndicatorShape indicatorShape;
  bool lastPage = false;

  _GetStartedState(this.pageList, this.pg, this.indicatorShape,
      this.onFinishPress, this.onSkipPress);

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        backgroundColor: hexToColor(pg.bgColor),
        body: PageIndicatorContainer(
          pageView: PageView(
            controller: pgController,
            children: _createChildren(),
            onPageChanged: (int index) {
              setState(() {
                lastPage = index == pageList.length - 1 ? true : false;
              });
            },
          ), //PageView
          length: pageList.length,
          indicatorColor: hexToColor(pg.indicatorColor),
          indicatorSelectorColor: hexToColor(pg.indicatorSelectorColor),
          shape: indicatorShape,
        ), //PageIndicatorContainer

        bottomNavigationBar: Padding(
            padding: EdgeInsets.only(left: 10.0, right: 10.0),
            child: Row(
              children: <Widget>[
                Expanded(
                  child: ListTile(
                      title: Text(
                        pg.skipButtonText,
                        textAlign: TextAlign.center,
                        style: TextStyle(
                            color: hexToColor(pg.skipButtonColor),
                            fontFamily: pg.skipButtonFont,
                            fontSize: pg.skipButtonSize),
                      ),
                      onTap: () {
                        onSkipPress();
                      }),
                ),
                Expanded(
                    child: lastPage
                        ? ListTile(
                            title: Text(
                              pg.lastNextButton,
                              textAlign: TextAlign.center,
                              style: TextStyle(
                                  color: hexToColor(pg.lastButtonColor),
                                  fontFamily: pg.lastButtonFont,
                                  fontSize: pg.lastButtonSize),
                            ),
                            onTap: () {
                              onFinishPress();
                            },
                          )
                        : ListTile(
                            title: Text(
                              pg.nextButtonText,
                              textAlign: TextAlign.center,
                              style: TextStyle(
                                  color: hexToColor(pg.nextButtonColor),
                                  fontFamily: pg.nextButtonFont,
                                  fontSize: pg.nextButtonSize),
                            ),
                            onTap: () {
                              setState(() {
                                pgController.nextPage(
                                    duration: Duration(milliseconds: 500),
                                    curve: Cubic(0.5, 0.5, 0.5, 0.5));
                              });
                            },
                          ))
              ],
            )) //Padding,

        );
  } //build()

  List<Widget> _createChildren() {
    return new List<Widget>.generate(pageList.length, (int index) {
      return getView(
          pageList[index].pageColor,
          pageList[index].title,
          pageList[index].titleColor,
          pageList[index].titleFont,
          pageList[index].titleSize,
          pageList[index].description,
          pageList[index].descriptionColor,
          pageList[index].descriptionFont,
          pageList[index].descriptionSize,
          pageList[index].imageIcon);
    });
  }

  Color hexToColor(String code) {
    return new Color(int.parse(code.substring(1, 7), radix: 16) + 0xFF000000);
  }

  Card getView(
      String pageColor,
      String title,
      String titleColor,
      String titleFont,
      double titleSize,
      String description,
      String descriptionColor,
      String descriptionFont,
      double descriptionSize,
      Widget icon) {
    return Card(
      color: hexToColor(pageColor),
      elevation: 12.0,
      margin: EdgeInsets.all(30.0),
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12.0)),
      child: Padding(
          padding: EdgeInsets.all(20.0),
          child: Column(
            mainAxisSize: MainAxisSize.min,
            children: <Widget>[
              Center(
                child: SizedBox(
                  child: Text(
                    "$title",
                    style: TextStyle(
                        fontSize: titleSize,
                        color: hexToColor(titleColor),
                        fontFamily: titleFont),
                  ),
                  height: 70.0,
                ),
              ),
              Expanded(
                  child: Padding(
                      padding: EdgeInsets.all(10.0),
                      child: SingleChildScrollView(
                        child: Text(
                          "$description",
                          style: TextStyle(
                              fontSize: descriptionSize,
                              color: hexToColor(descriptionColor),
                              fontFamily: descriptionFont),
                          textAlign: TextAlign.justify,
                        ),
                        //height: 200.0,
                      ))),
              Padding(
                padding: EdgeInsets.all(10.0),
                child: SizedBox(height: 100.0, child: icon),
              )
            ],
          )),
    ); //Card
  }
}


walkthrough/lib/classes/page.dart
class Page {
  final String bgColor;
  final String indicatorColor;
  final String indicatorSelectorColor;
  final String skipButtonText;
  final String skipButtonFont;
  final String skipButtonColor;
  final double skipButtonSize;
  final String nextButtonText;
  final String nextButtonFont;
  final String nextButtonColor;
  final double nextButtonSize;
  final String lastButtonFont;
  final String lastButtonColor;
  final double lastButtonSize;
  final String lastNextButton;

  Page({
    this.bgColor = "#ffffff",
    this.indicatorColor = "#ffffff",
    this.indicatorSelectorColor = "#195599",
    this.skipButtonText = "SKIP",
    this.skipButtonColor = "#195599",
    this.skipButtonFont = "Lato",
    this.skipButtonSize = 20.0,
    this.nextButtonColor = "#195599",
    this.nextButtonFont = "Lato",
    this.nextButtonSize = 20.0,
    this.nextButtonText = "NEXT",
    this.lastNextButton = "FINISH",
    this.lastButtonColor = "#1f8e2f",
    this.lastButtonFont = "Lato",
    this.lastButtonSize = 20.0,
  });
}


walkthrough/example/lib/example.dart
import 'package:flutter/material.dart';
import 'package:get_started/classes/walkthrough.dart';
import 'package:get_started/classes/page.dart';
import 'package:get_started/get_started.dart';
import 'package:page_indicator/page_indicator.dart';

void main() {
  runApp(new MyApp1());
}

class MyApp1 extends StatefulWidget {
  //All the properties defined for 'page' have some default values. Hence, you may skip some size, etc. if you wish to!
  @override
  _MyApp1State createState() => _MyApp1State();
}

class _MyApp1State extends State<MyApp1> {
  Page pageObj = new Page(
      bgColor: "#ffffff",
      indicatorColor: "#ffffff",
      indicatorSelectorColor: "#195599",
      nextButtonText: "NEXT",
      nextButtonColor: "#195599",
      nextButtonFont: "Lato",
      nextButtonSize: 20.0,
      skipButtonText: "SKIP",
      skipButtonFont: "Lato",
      skipButtonColor: "#195599",
      skipButtonSize: 20.0,
      lastNextButton:
          "FINISH", //You may also write "GOT IT" or whatever you wish :)
      lastButtonColor: "#1f8e2f",
      lastButtonFont: "Lato",
      lastButtonSize: 20.0);

  List<Walkthrough> pageList = [
    Walkthrough(
        pageColor: "#48C9B0",
        title: "Welcome!",
        titleFont:
            "Lato", //Don't forget to include yours fonts in pubspec.yaml file.
        titleColor: "#ffffff", //Pass the colours as strings of their hex codes.
        titleSize: 25.0,
        description:
            "This is a Scrollable Text Widget. Hence, don't worry if your description goes long. :) \n\nA Page has three sections for Title, Description and an Icon respectively that can be customized for the text, font, color and size.\n\nAll the properties defined for 'walkthrough' have some default values. Hence, you may skip some size, etc. if you wish to!",
        descriptionFont: "Lato",
        descriptionColor: "#ffffff",
        descriptionSize: 16.0,
        imageIcon: Icon(
          Icons.computer,
          color: Colors.white,
          size: 100.0,
        )),
    Walkthrough(
        pageColor: "#EC7063",
        title: "Add a photo!",
        description: "Building your memory lane with your photographs. :)",
        imageIcon: Icon(
          Icons.add_a_photo,
          color: Colors.white,
          size: 100.0,
        )),
    Walkthrough(
        pageColor: "#F4D03F",
        title: "Documents",
        description: "Store your files safe and sound here.",
        imageIcon: Icon(
          Icons.description,
          color: Colors.white,
          size: 100.0,
        )),
  ];

  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(
          title: Text(
            "Getting Started",
            style: TextStyle(fontFamily: "Lato"),
          ),
          backgroundColor: Color(0xff195599),
        ), //appBar

        //Required Field
        //body: GetStarted(pageList : pageList),

        //Required Field with Optional parameters
        body: GetStarted(
            pageList: pageList,
            pageStyle: pageObj,
            indicatorShape: IndicatorShape.circle(size: 12.0),
            onFinishPress: _finishPress,
            onSkipPress: _skipPress),
      ),
    );
  }

  void _skipPress() {
    //define your skipButton functionality here.
  }

  void _finishPress() {
    //define your lastNextButton functionality here.
  }
}
